USECASES.md

```diff
@@ -1,5 +1,23 @@
 # Use Cases
 
+## Provisioning Models
+
+**Network Peripheral**: Peripheral to the CPU exposing a network interface to the CPU OS, provisioned directly by the CPU OS.
+<br>
+![Network Peripheral](NetworkPeripheral.svg)
+<br>
+**Data Processor Peripheral**: Peripheral to the CPU exposing a data processor (DPU) interface to the CPU OS, provisioned by the platform BMC.
+<br>
+![Data Processing Peripheral](DataProcessingPeripheral.svg)
+<br>
+**Infrastructure Device**: Independent device from the CPU exposing virtual devices to the CPU OS, provisioned by a trusted software entity over the network.
```

Comment:
This seems to overlap with the Data Processor Peripheral as described.  They both essentially expose a DPU interface to the CPU (or can expose that).  The difference is the provisioning control point.

Perhaps we can consolidate these down to 2 to 3 types of devices and adjust the names as they add to the confusion.

Reply:
Hi which ones are overlapping?  The DataProcessingPeripheral is very different from an InfrastructureDevice since one is controlled by the host and the other is an independent device.  This is a fundamental difference for a lot of things especially security.

Reply:
The Infrastructure Device and Data Processing Peripheral seem to overlap (at least from that the text describes and implies.  Both basically need the same functions including security it just seems that the difference is that one is an independent device and the other is inside of a platform with a dependency on power.  Maybe this is just my confusion in reading of the descriptions.

Reply:
That is the main difference, which is pretty fundamental.

For the sake of provisioning, we could even reduce this down to two models:

 - HostPeripheral:  Provisioned by the host
 - InfrastructureDevice:  Provisioned by the infrastructure
 
This simpler modeling also ignores the details around to BMC or not BMC, since this is more of how you implement that host-based or infrastructure-based provisioning.

If this simplification makes sense- what I can do is close this pull request and do a new one with only two models.

Reply:
@dandaly and `HostPeripheral` is not an interesting use case, since this is already handled by all major server companies like Dell, HP and others ? or there is still a gap we need to address ? either platform BMC using redfish or Host OS iunband will provision those, correct ?

Reply:
First we should identify the Provisioning Models, and then if we want to say which ones are in scope or out of scope that would be a second step.  I think having two models identified, and saying that OPI is focused on enabling just one of them would be a great result.